gitter/score.py

```diff
@@ -0,0 +1,62 @@
+from gitter.splitter import Splitter
+from gitter.pos_tagger import POSTagger
+from gitter.dict_tagger import DictionaryTagger
+
+
+def tagged_sentences(text):
+    """
+    Tag a message with dict_tagged_sentence.
+
+    This method takes a string parameter which is the message
+    send by the newcomer and then it split it into list of lists
+    of words using the Splitter class, and it uses POSTagger class
+    to tag each word with pos. And then it uses the DictionaryTagger
+    class and the two dict 'gitter/question.yml' and 'gitter/ignore.yml'
+    and tag the pos tagged word with 'question' or 'ignore'.
+
+    :param text: a string representing the message.
+                 e.g. 'How to solve this issue?'
+    :return: a list of words tagged with DictTagger and POSTagger.
+             e.g. [[('how', 'how', ['question', 'WRB']),
+                    ('can', 'can', ['MD']),
+                    ('I', 'I', ['PRP']),
+                    ('solve', 'solve', ['VB']),
+                    ('this', 'this', ['DT']),
+                    ('issue', 'issue', ['NN']),
+                    ('?', '?', ['question', '.'])]]
+    """
+    splitter = Splitter()
+    postagger = POSTagger()
+    splitted_sentences = splitter.split(text)
+    pos_tagged_sentences = postagger.pos_tag(splitted_sentences)
+    dicttagger = DictionaryTagger(['gitter/question.yml', 'gitter/ignore.yml'])
+    dict_tagged_sentences = dicttagger.tag(pos_tagged_sentences)
+    return dict_tagged_sentences
+
+
+def value_of(sentiment):
+    """
+    Get a value based on the sentiment.
+
+    :param sentiment: a string representing the sentiment of
+                      a word. i.e. 'ignore' or 'question'.
+    :return: an integer representing the value of the sentiment.
+    """
+    if sentiment == 'ignore':
+        return 1
+    if sentiment == 'question':
+        return -1
+    return 0
```

Comment:
this system currently only knows about two sentiments, ignore and question (answer is missing).

It should raise a ValueError here if the sentiment is not known, as its score can not be determined.

Reply:
@jayvdb, actually the strategy I am using here is a message which don't include the `ignore` and `question` words should be categorized into `answer`. That's why I am returing `0` here, so that when we get the `sentiment_score=0`, we can say that this message is a answer.